Include/internal/pycore_stackref.h

```diff
@@ -190,9 +189,15 @@ static const _PyStackRef PyStackRef_NULL = { .bits = 0 };
 
 #endif // Py_GIL_DISABLED
 
-// Note: this is a macro because MSVC (Windows) has trouble inlining it.
+// Check if a stackref is exactly the same as another stackref, including the
+// the deferred bit. This can only be used safely if you know that the deferred
+// bits of `a` and `b` match.
```

Comment:
This seems like it might be easy to accidentally get wrong. Would it be worth adding a different implementation of `PyStackRef_IsExactly` in debug builds that asserts that the deferred bits of `a` and `b` match?